mujoco_ros/src/mushr_control_listener.cpp

```diff
@@ -43,14 +48,14 @@ class MushrControlListener
     save_subscriber = n.subscribe("save_trajectory", 1000, &MushrControlListener::save_callback, this);
   }
 
-  void control_callback(const mujoco_ros::MushrControl::ConstPtr& msg)
+  void control_callback(const mj_models::MushrPlan::ConstPtr& msg)
   {
-    std::vector<double> control;
-    control.push_back(msg->steering_angle.data);
-    control.push_back(msg->velocity.data);
-    sim->set_control(control);
-    sim->propagate(msg->duration.data);
-    ROS_INFO("Result: %f, %f", sim->d->qpos[0], sim->d->qpos[1]);
+    for (int i = 0; i < msg->control.size(); ++i)
+    {
+      mj_models::copy(sim->d->ctrl, msg->control[i]);
```

Comment:
This would create a circular dependency between mj_models and mujoco-ros due to `mujoco_simulator_t`. In another PR I added a [templated control listener ](https://github.com/PRX-Kinodynamic/mujoco_ros/blob/rt-viz/mujoco_ros/include/mujoco_ros/control_listener.hpp) that handles the `sim->d->ctrl`.

Also, the current `mj_models::copy` allows to copy to any subscribable object, not only a `mujoco_simulator_t`.

Reply:
SGTM.